designs/mixins.md

```diff
@@ -692,12 +698,88 @@ The members are ordered as follows:
 - `sizeFilter`
 
 
+### Mixins on shapes with non-member properties
+
+Some shapes don't have members, but do have other properties. Adding a mixin
+to such a shape causes the properties of the other shape to be merged into
+the shape. Scalar properties defined in the local shape are kept, and
+non-scalar properties are merged. When merging map properties, the values for
+local keys are kept. The ordering of merged lists / sets follows the
+same ordering as members.
+
+For example, in the following model:
+
+```
+operation OperationA {}
+
+service A {
+    version: "A"
+    operations: [OperationA]
+}
+
+operation OperationB {}
+
+service B with A {
+    version: "B"
+    rename: {
+        "smithy.example#OperationA": "OperA"
+        "smithy.example#OperationB": "OperB"
+    }
+    operations: [OperationB]
+}
+
+operation OperationC {}
+
+service C with B {
+    version: "C"
+    rename: {
+        "smithy.example#OperationA": "OpA"
+        "smithy.example#OperationC": "OpC"
+    }
+    operations: [OperationC]
+}
+```
+
+The `version` property of the local shape is kept and the `rename` and
+`operations` properties are merged. This is equivalent to the following:
+
+```
+operation OperationA {}
+
+operation OperationB {}
+
+operation OperationC {}
+
+service C {
+    version: "C"
+    rename: {
+        "smithy.example#OperationA": "OpA"
+        "smithy.example#OperationB": "OperB"
+        "smithy.example#OperationC": "OpC"
+    }
+    operations: [OperationA, OperationB, OperationC]
+}
+```
+
+### Resource mixins
+
+Resource shapes with the `@mixin` trait MAY NOT define any properties. This is
```

Comment:
Wouldn’t defining identifiers be ok? You could then extend the identifiers map in shapes that use the mixin. This might make it easier to define resource hierarchies.

Reply:
I'm not sure what this makes easier. Resource hierarchies are modeled using the `resources` property.